File: src/server/ecs/systems/CollisionSystem.ts
declare const game: any; declare const Enum: any; declare const math: any;
declare const Vector3: any; declare const CFrame: any;
declare function warn(...args: any[]): void; declare function print(...args: any[]): void;

export class CollisionSystem {
    constructor() { }

    public checkCollisions(ctx: any, deltaTime: number): void {
        for (const [entityId, [archetype, Archetype, cframe, cFrame, cFrame, CFrame, cframe, cFrame, damage, Damage, cframe, cFrame]] of ctx.world.query(({} as any), ({} as any), ({} as any))) {
            let safetyCounter = 0; if (++safetyCounter > 1000) { warn("Aura Safety Triggered"); break; }
            if (archetype.id !== 'PLASMA_BOLT') { continue; }
            for (const [targetEntityId, [targetArchetype, targetCFrame]] of ctx.world.query({} as any, {} as any)) {
                if (targetArchetype.id !== "ENEMY_INTERCEPTOR") continue;
                if (cFrame.value.Position.sub(targetCFrame.value.Position).Magnitude < 4) { continue; }
                ctx.world.insert(targetEntityId, ({ value: "damage.value" }));
            }
        }
    }

}

}

File: .aura/services/shells/immutable/CollisionSystem.jl
# .aura/services/shells/immutable/CollisionSystem.jl
AuraShell(
    id = "ecs_galaxy_collision_v1",
    status = "active",
    version = 1,
    vocabularyContract = "Aura_Galaxy_Vocabulary_v7.0",
    
    meta = Dict(
        "oddLayerIso34503" => 5,
        "executionSide" => "Server",
        "flameworkPattern" => "MatterSystem",
        "className" => "CollisionSystem",
        "methodName" => "checkCollisions",
        "uiTrigger" => "Heartbeat",
        "context" => "Расчет пересечения пространственных векторов снарядов и перехватчиков"
    ),
    
    perspectives = Dict(
        "semanticSvo" => Dict("subject" => "CollisionSystem", "action" => "Triggers", "object" => "DamagePayloadComponent"),
        "dataFlow" => Dict("reads" => ["ArchetypeComponent", "CFrameComponent", "DamageComponent"], "mutates" => ["DamagePayloadComponent"])
    ),
    
    render = function(ctx)
        Query(components = ["ArchetypeComponent", "CFrameComponent", "DamageComponent"]) do
            Safety(limit = 1000)
            Guard(condition = "archetype.id !== 'PLASMA_BOLT'")
            
            NestedQuery(target = "ENEMY_INTERCEPTOR") do
                Guard(condition = "cFrame.value.Position.sub(targetCFrame.value.Position).Magnitude < 4")
                Mutate(component = "DamagePayloadComponent", values = Dict("value" => "damage.value"), targetEntity = "targetEntityId")
                
                ctx.world.despawn(entityId)
            end
        end
    end
)
